loopd/swapclient_server.go

```diff
@@ -738,6 +739,74 @@ func (s *swapClientServer) GetLsatTokens(ctx context.Context,
 	return &clientrpc.TokensResponse{Tokens: rpcTokens}, nil
 }
 
+// GetInfo returns basic information about the loop daemon and details to swaps
+// from the swap store.
+func (s *swapClientServer) GetInfo(_ context.Context,
+	_ *clientrpc.GetInfoRequest) (*clientrpc.GetInfoResponse,
+	error) {
+
+	// Fetch loop-outs from the loop db.
+	outSwaps, err := s.impl.Store.FetchLoopOutSwaps()
+	if err != nil {
+		return nil, err
+	}
+
+	// Collect loop-out stats.
+	loopOutStats := &clientrpc.LoopOutStats{}
+	for _, out := range outSwaps {
+		switch out.State().State.Type() {
+		case loopdb.StateTypeSuccess:
+			loopOutStats.SuccessCount++
+			loopOutStats.SumSucceededAmt += int64(
+				out.Contract.AmountRequested,
+			)
+		case loopdb.StateTypePending:
+			loopOutStats.PendingCount++
+			loopOutStats.SumPendingAmt += int64(
+				out.Contract.AmountRequested,
+			)
+		case loopdb.StateTypeFail:
+			loopOutStats.FailCount++
+		}
+	}
+
+	// Fetch loop-ins from the loop db.
+	inSwaps, err := s.impl.Store.FetchLoopInSwaps()
+	if err != nil {
+		return nil, err
+	}
+
+	// Collect loop-in stats.
+	loopInStats := &clientrpc.LoopInStats{}
+	for _, in := range inSwaps {
```

Comment:
OK update here:
The `.Contract` part that `out` and `in` swaps access seems to be of a different type (`LoopInContract` and `LoopOutContract`) so you can't handle them over a common func, which would lead to new interfaces etc... overkill IMO :)

closing this